sklearn/decomposition/tests/test_pca.py

```diff
@@ -112,10 +112,8 @@ def test_explained_variance():
 
     pca = PCA(n_components=2).fit(X)
     rpca = RandomizedPCA(n_components=2, random_state=42).fit(X)
-    assert_array_almost_equal(pca.explained_variance_,
-                              rpca.explained_variance_, 1)
     assert_array_almost_equal(pca.explained_variance_ratio_,
-                              rpca.explained_variance_ratio_, 3)
+                              rpca.explained_variance_ratio_, 1)
```

Comment:
The test passed with 3 power iterations, the default for `RandomizedPCA` before. With only 2 I had to reduce the precision.

I have added a test on correlated data and indeed it works with higher precision.